src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -90,8 +91,283 @@ public void testOffsetAtReference(String cigar, int posInReference, int expected
 
     @Test(dataProvider = "referenceAtReadData")
     public void testOffsetAtRead(String cigar, int expectedReferencePos, int posInRead) {
+        final SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        Assert.assertEquals(sam.getReferencePositionAtReadPosition(posInRead), expectedReferencePos);
+    }
 
-            SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
-            Assert.assertEquals(sam.getReferencePositionAtReadPosition(posInRead), expectedReferencePos);
+    @Test
+    public void testReferenceName() {
+        // NO_ALIGNMENT_NAME
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, "3S9M", null, 2);
```

Comment:
Extract this `new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, "3S9M", null, 2);` call into a helper method, and call it here and everywhere else this line is duplicated.